src/sharepoint/order/sharepoint-order.controller.js

```diff
@@ -103,98 +125,74 @@ angular
-          _.set(account, 'activateSharepoint', _.some(this.accountsToActivate, account.primaryEmailAddress));
-          return account;
+          primaryEmailAddress: email,
         });
     }
 
-    onTranformItemDone() {
-      this.loaders.accounts = false;
-    }
-
-    checkSharepointActivation(account) {
-      if (account.activateSharepoint) {
-        this.accountsToActivate.push(account.primaryEmailAddress);
-      } else {
-        this.accountsToActivate
-          .splice(this.accountsToActivate.indexOf(account.primaryEmailAddress), 1);
+    onAssociateChange(associateExchange) {
+      if (!associateExchange) {
+        // we need to reset selected accounts if we leave the association interface
+        this.accountsToAssociate = [];
       }
     }
 
-    getExchangeAccountsUrl() {
-      return `#/configuration/exchange_hosted/${this.associatedExchange.organization}/${this.associatedExchange.name}?tab=ACCOUNTS`;
-    }
-
-    static hasInexistantAccount(activateSharepointForm) {
-      return activateSharepointForm.primaryEmailAddressField.$invalid
-        && !_.isEmpty(activateSharepointForm.primaryEmailAddressField.$viewValue);
+    refreshAccounts(exchange) {
+      this.associatedExchange = exchange;
+      this.ouiDatagridService.refresh('exchangeAccountsDatagrid');
     }
 
-    changeStandAloneQuantity() {
-      if (parseInt(this.standAloneQuantity, 10) >= 1
-        && parseInt(this.standAloneQuantity, 10) <= 30) {
-        this.standAloneQuantity = parseInt(this.standAloneQuantity, 10);
-      }
+    onAccountsSelected(accounts) {
+      this.accountsToAssociate = accounts.map(({ primaryEmailAddress }) => primaryEmailAddress);
     }
 
-    decrement() {
-      if (this.standAloneQuantity > 1) {
-        this.standAloneQuantity -= 1;
-      }
-    }
-
-    increment() {
-      if (this.standAloneQuantity < 30) {
-        this.standAloneQuantity += 1;
-      }
+    hasSelectedAccounts() {
+      return this.accountsToAssociate.length > 0;
     }
 
     getSharepointOrderUrl() {
       if (this.associateExchange) {
-        if (_.has(this.associatedExchange, 'name') && this.accountsToActivate.length >= 1) {
-          return this.sharepointService.getSharepointOrderUrl(
+        if (_.has(this.associatedExchange, 'name') && this.accountsToAssociate.length >= 1) {
+          return this.Sharepoint.getSharepointOrderUrl(
             this.associatedExchange.name,
-            this.accountsToActivate,
+            this.accountsToAssociate,
           );
         }
         return '';
       }
-      if (!_.isNull(this.standAloneQuantity)
-        && parseInt(this.standAloneQuantity, 10) >= 1
-        && parseInt(this.standAloneQuantity, 10) <= 30) {
-        return this.sharepointService
-          .getSharepointStandaloneOrderUrl(parseInt(this.standAloneQuantity, 10));
+
+      const quantity = parseInt(this.standAloneQuantity, 10);
+      if (quantity >= 1 && quantity <= 30) {
```

Comment:
So if the quantity is invalid you don't do anything ? Why do you allow invalid values ?

Reply:
We don't allow invalid values. It was just some extra checks left from the old code. I prefer to keep it at least once (the check was done about 4 times, I reduced it to 1).